lib/ret.ex

```diff
@@ -16,4 +21,76 @@ defmodule Ret do
       {:error, :forbidden}
     end
   end
+
+  defp delete_avatars_for_account(%Account{} = account) do
+    account_avatars_query = from(avatar in Avatar, where: avatar.account_id == ^account.account_id)
+    account_avatars = Repo.all(account_avatars_query) |> Repo.preload(Avatar.file_columns())
+
+    account_avatar_owned_files =
+      account_avatars
+      |> Enum.flat_map(fn avatar ->
+        Avatar.file_columns()
+        |> Enum.map(fn column -> Map.fetch!(avatar, column) end)
+        |> Enum.filter(fn owned_file -> owned_file != nil end)
```

Comment:
Nested pipelines are difficult to read. The anonymous function given to `Enum.flat_map/2` is a strong contender for a private function.

Additionally the map and filter calls make this a good opportunity for a list comprehension.

```elixir
defp avatar_owned_files(avatar) do
  for column <- Avatar.file_columns(),
      owned_file = Map.fetch!(avatar, column),
      do: owned_file
end
```

The first line of the comprehension is the generator. The second line is the filter. It rejects any falsey values. We bind in the filter so we can access the value in the final line.